### DISCUSSION TITLE
Support for custom checkers

### DISCUSSION THREAD

**Discussion by yarikoptic** [upptime/upptime#948] [Ideas]
ATM it is `upptime`'s duty to do the check, and check has only some limited set of configuration parameters available.
To allow for flexibility and also rapid workarounds for e.g. time points while PRs like 

- https://github.com/upptime/uptime-monitor/pull/245

are reviewed/merged/released, it would be great if it was possible to configure external check scripts which would provide result of the check as some structured record. E.g. it could be the configuration

```yaml
- name: Custom check
  external_check: scripts/check_my_service
  assignees:
      - yarikoptic
```

where `scripts/check_my_service` is an executable (on linux) application/script which would return a json record like

```
{
    "status": "ok",
    "return_code": 200,
    "response_time": 0.123,
    "details": "all is good"
}
```
where `status` is mandatory and could take values

- "ok": all good
- "error": if some non-ok result
- "internal-error": internal error of the checker

etc.  That would allow for easy crafting of custom checkers to extend the wonderful land of upptime reports!

**Comment by NilsBaumgartner1994** [comment]
Would toatlly love this !

could be easily to integrate:

```response.data.details === "all is good"```
